SC_redesign/Crop_and_Soil/soil/nitrogen_cycling/denitrification.py

```diff
@@ -0,0 +1,80 @@
+from typing import Optional
+from math import exp
+
+from SC_redesign.Crop_and_Soil.soil.soil_data import SoilData
+
+"""
+This module handles the denitrification of nitrogen in the nitrates pool, based on SWAT section 3:1.4.
+"""
+
+
+class Denitrification:
+
+    def __init__(self, soil_data: Optional[SoilData] = None, field_size: Optional[float] = None):
+        """This method initializes the SoilData object that this module will work with, or create one if none provided.
+
+        Parameters
+        ----------
+        soil_data : SoilData, optional
+            The SoilData object used by this module to track the denitrification of nitrates in the soil profile,
+            creates new one if one is not provided.
+        field_size : float, optional
+            Size of the field (ha)
+
+        Notes
+        -----
+        The field size is used to initialize a SoilData object for this module to work with, if a pre-configured
+        SoilData object is not provided.
+
+        """
+        self.data = soil_data or SoilData(field_size=field_size)
+
+    def do_denitrification(self) -> None:
+        """Conducts the daily denitrification operations.
+
+        References
+        ----------
+        SWAT Theoretical documentation section 3:1.4, eqn. 3:1.4.1, 2
+
+        """
+        for layer in self.data.soil_layers:
+            if layer.nutrient_cycling_water_factor < layer.denitrification_threshold_water_content:
+                continue
+
+            # TODO: check that soil_overall_carbon_fraction is being used correctly in this context - issue #475
+            nitrified_nitrates = self._calculate_nitrification_amount(layer.nitrate_content,
+                                                                      layer.denitrification_rate_coefficient,
+                                                                      layer.nutrient_cycling_temp_factor,
+                                                                      layer.soil_overall_carbon_fraction * 100)
+            layer.nitrate_content -= nitrified_nitrates
```

Comment:
Does this need to be bounded to 0? I'm genuinely asking.

```suggestion
            layer.nitrate_content -= max(nitrified_nitrates, 0)
```

If so, perhaps the bounding should be part of the return statement for `_calculate_nitrification_amount()` instead of here.

Reply:
I agree, I will make that change to `_calculate_nitrification_amount()`